modules/core/src/test/java/org/apache/synapse/config/xml/BeanMediatorFactoryTest.java

```diff
@@ -0,0 +1,109 @@
+
+/**
+ * This is the test class for BeanMediatorFactory class.
+ */
+public class BeanMediatorFactoryTest {
+
+    /**
+     * Creating a bean mediator and asserting it is created successfully.
+     *
+     * @throws XMLStreamException - XMLStreamException.
+     */
+    @Test
+    public void testCreateSpecificMediator() throws XMLStreamException {
+        BeanMediatorFactory beanMediatorFactory = new BeanMediatorFactory();
+        String inputXML = "<bean action=\"CREATE\" class=\"org.apache.synapse.util.TestTask\" " +
+                "var=\"loc\"></bean>";
+        OMElement element = AXIOMUtil.stringToOM(inputXML);
+        Mediator mediator = beanMediatorFactory.createSpecificMediator(element, null);
+        Assert.assertTrue("BeanMediator is not created successfully.", mediator instanceof
+                BeanMediator);
+        BeanMediator beanMediator = (BeanMediator) mediator;
+        Assert.assertEquals("CREATE action is not set", BeanMediator.Action.CREATE, beanMediator.getAction());
+    }
+
+    /**
+     * Setting property of BeanMediator and asserting the action.
+     *
+     * @throws XMLStreamException - XMLStreamException.
+     */
+    @Test
+    public void testCreateSpecificMediator2() throws XMLStreamException {
+        BeanMediatorFactory beanMediatorFactory = new BeanMediatorFactory();
+        String inputXML = "<bean action=\"SET_PROPERTY\" property=\"latitude\" value=\"{//latitude}\" var=\"loc\" " +
+                "xmlns:ns3=\"http://org.apache.synapse/xsd\"             " +
+                "xmlns:ns=\"http://org.apache.synapse/xsd\"></bean>\n";
+        OMElement element = AXIOMUtil.stringToOM(inputXML);
+        Mediator mediator = beanMediatorFactory.createSpecificMediator(element, null);
+        Assert.assertTrue("BeanMediator is not created successfully.", mediator instanceof
+                BeanMediator);
+        BeanMediator beanMediator = (BeanMediator) mediator;
+        Assert.assertEquals("SET_PROPERTY action is not set", BeanMediator.Action.SET_PROPERTY, beanMediator.getAction());
+    }
+
+    /**
+     * Getting the property of BeanMediator and asserting the action.
+     *
+     * @throws XMLStreamException - XMLStreamException.
+     */
+    @Test
+    public void testCreateSpecificMediator3() throws XMLStreamException {
+        BeanMediatorFactory beanMediatorFactory = new BeanMediatorFactory();
+        String inputXML = "<bean action=\"GET_PROPERTY\" property=\"latitude\" value=\"{//latitude}\" var=\"loc\" " +
+                "target=\"target\"   xmlns:ns3=\"http://org.apache.synapse/xsd\"             " +
+                "xmlns:ns=\"http://org.apache.synapse/xsd\"></bean>\n";
+        OMElement element = AXIOMUtil.stringToOM(inputXML);
+        Mediator mediator = beanMediatorFactory.createSpecificMediator(element, null);
+        Assert.assertTrue("BeanMediator is not created successfully.", mediator instanceof
+                BeanMediator);
```

Comment:
Shall we correct this line segment as follows, to make it more readable?

Reply:
Fixed with https://github.com/wso2/wso2-synapse/pull/1053